src/ServiceInsight/MessageList/MessageListView.xaml.cs

```diff
@@ -102,5 +106,25 @@ void OnGridClicked(object sender, MouseButtonEventArgs e)
                 Model.BringIntoView(Model.Selection.SelectedMessage);
             }
         }
+
+        public void OnSaveLayout(ISettingsProvider settingsProvider)
+        {
+            var layoutSetting = settingsProvider.GetSettings<MessageListSettings>();
+            layoutSetting.GridLayout = grid.GetLayout();
+            settingsProvider.SaveSettings(layoutSetting);
+        }
+
+        public void OnRestoreLayout(ISettingsProvider settingsProvider)
+        {
+            var layoutSetting = settingsProvider.GetSettings<MessageListSettings>();
+            grid.RestoreLayout(layoutSetting.GridLayout.GetAsStream());
+        }
+
+        public void OnResetLayout(ISettingsProvider settingsProvider)
+        {
+            var layoutSettings = settingsProvider.GetSettings<MessageListSettings>();
+            layoutSettings.GridLayout = null;
+            settingsProvider.SaveSettings(layoutSettings);
```

Comment:
Why does nothing happen with `grid` in this method?

Reply:
This is the Reset method, which would null-out the stored layout in the settings (so it falls back to the default layout). This setting - generally speaking, all settings - become effective on the next restart (we have a UI hint for it as well).